File: gradle-groovy/sample/src/test/java/com/github/igabaydulin/sample/AppTest.java
package com.github.igabaydulin.sample;

import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

import static org.junit.jupiter.api.Assertions.*;

class AppTest {

    private PrintStream systemOutput = System.out;
    private ByteArrayOutputStream output = new ByteArrayOutputStream();

    @BeforeEach
    void before() {
        System.setOut(new PrintStream(output));
    }

    @AfterEach
    void after() {
        System.setOut(systemOutput);
    }

    @Test
    void testAppHasPing() {
        App.main(new String[]{});
        assertEquals("ping\n", output.toString());
    }
}

File: gradle-kotlin/sample/src/test/kotlin/com/github/igabaydulin/sample/AppTest.kt
package com.github.igabaydulin.sample

import org.junit.After
import org.junit.Before
import kotlin.test.Test
import kotlin.test.assertNotNull
import kotlin.test.assertTrue
import java.io.PrintStream
import java.io.ByteArrayOutputStream
import kotlin.test.assertEquals

class AppTest {

    private val systemOutput = System.out
    private val output = ByteArrayOutputStream()

    @Before
    fun before() {
        System.setOut(PrintStream(output))
    }

    @After
    fun after() {
        System.setOut(systemOutput)
    }

    @Test
    fun testAppHasPing() {
        main()
        assertEquals("ping\n", output.toString())
    }
}
